validation/open_boundaries/mode_1_internal_wave.jl

```diff
@@ -0,0 +1,108 @@
+using Oceananigans
+using Oceananigans.Fields: FunctionField
+
+# TODO: clean up all the global variables in this script
+
+h = 10
+k = π/h
+
+λ = 2π / k
+
+grid = RectilinearGrid(topology = (Bounded, Flat, Bounded), size = (256, 128), extent = (3*λ, h))
+
+xc, yc, zf = nodes(grid, Center(), Center(), Face())
+xf, yc, zc = nodes(grid, Face(), Center(), Center())
+
+N² = 0.2
+w₀ = .03
+
+ω = @show sqrt(k^2 * N² / (k^2 + (π/h)^2))
+
+u(x, z, t) = w₀ * π / (k * h) * cos(π * z / h) * cos(k * x - ω * t)
+
+w(x, z, t) = w₀ * sin(π * z / h) * sin(k * x - ω * t)
+
+b(x, z, t) = - N² * w₀ / ω * sin(π * z / h) * cos(k * x - ω * t) + N² * z
+
+@inline u_east(z, t) = u(xf[grid.Nx], z, t)
+@inline u_west(z, t) = u(0, z, t)
+
+@inline w_east(z, t) = w(xc[grid.Nx], z, t)
+@inline w_west(z, t) = w(0, z, t)
+
+@inline b_east(z, t) = b(xc[grid.Nx], z, t)
+@inline b_west(z, t) = b(0, z, t)
+
+u_boundaries = FieldBoundaryConditions(east = OpenBoundaryCondition(u_east),
+                                       west = OpenBoundaryCondition(u_west))
+
+w_boundaries = FieldBoundaryConditions(east = OpenBoundaryCondition(w_east), 
+                                       west = OpenBoundaryCondition(w_west)) 
+
+b_boundaries = FieldBoundaryConditions(east = OpenBoundaryCondition(b_east),
+                                       west = OpenBoundaryCondition(b_west))
+
+sponge_thickness = λ/2
+
+mask(x, z) = ((sponge_thickness - x) / sponge_thickness) * (x < sponge_thickness) + ((x - grid.Lx + sponge_thickness) / sponge_thickness) * (x > grid.Lx - sponge_thickness)
```

Comment:
I can't figure out what this is doing at a glance... it might help to rewrite it